Fix argocd for darwin arm

Jasstkn opened the pull request:
Signed-off-by: Maria Kotlyarevskaya <mariia.kotliarevskaia@gmail.com>

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
- fix the following error
```bash
arkade get argocd
Downloading: argocd
2022/10/15 11:54:32 Looking up version for argocd
2022/10/15 11:54:32 Found: v2.4.14
Downloading: https://github.com/argoproj/argo-cd/releases/download/v2.4.14/argocd-darwin-
Error: incorrect status for downloading tool: 404
```

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
- [ ] I have raised an issue to propose this change, which has been given a label of `design/approved` by a maintainer ([required](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md))


## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

```bash
./hack/test-tool.sh argocd
+ go build
+ ./arkade get argocd --arch arm64 --os darwin --quiet
+ file /Users/mariarti/.arkade/bin/argocd
/Users/mariarti/.arkade/bin/argocd: Mach-O 64-bit executable arm64
+ rm /Users/mariarti/.arkade/bin/argocd
+ ./arkade get argocd --arch x86_64 --os darwin --quiet
+ file /Users/mariarti/.arkade/bin/argocd
/Users/mariarti/.arkade/bin/argocd: Mach-O 64-bit executable x86_64
+ rm /Users/mariarti/.arkade/bin/argocd
+ ./arkade get argocd --arch x86_64 --os linux --quiet
+ file /Users/mariarti/.arkade/bin/argocd
/Users/mariarti/.arkade/bin/argocd: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=f1VgrqkqMGWN6_4dm8tE/45haly0gE5421RQcHydh/W2d2BfGZzTrmuzaht9nl/AVoIgXvoYE1TqXi1UALt, not stripped
+ rm /Users/mariarti/.arkade/bin/argocd
+ ./arkade get argocd --arch arm64 --os linux --quiet
+ file /Users/mariarti/.arkade/bin/argocd
/Users/mariarti/.arkade/bin/argocd: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), statically linked, Go BuildID=yLIHuhrd3pVkeBEpRr3a/Dl-2QIjY-48BJf2QsM-v/-5EFldwmS5MUEQUH3itC/fmznl87l7B22-ZiF2g65, not stripped
+ rm /Users/mariarti/.arkade/bin/argocd
+ ./arkade get argocd --arch x86_64 --os ming --quiet
+ file /Users/mariarti/.arkade/bin/argocd
/Users/mariarti/.arkade/bin/argocd: PE32+ executable (console) x86-64 (stripped to external PDB), for MS Windows
+ rm /Users/mariarti/.arkade/bin/argocd
```

## Are you a GitHub Sponsor yet (Yes/No?)

<!-- Requests from sponsors take priority -->
<!--- Check at https://github.com/sponsors/alexellis         -->

- [ ] Yes
- [x] No

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [ ] I have updated the list of tools in README.md if (required) with `./arkade get -o markdown`
- [ ] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

<!--- In case it is a new application -->
- [x] I have tested this on arm, or have added code to prevent deployment

alexellis reviewed (changes requested):
Can you add a test case to go with this change please?

Jasstkn commented:
> Can you add a test case to go with this change please?

Thank you for the review! Added test and refactored a little bit to be more consistent with the newer tests.

alexellis reviewed (approved):
Approved
